pages/category/index.vue

```diff
@@ -220,96 +190,99 @@
   name: "Category",
   components: {
     SfHeading,
     SfButton,
     SfIcon,
-    SfList,
-    SfMenuItem,
     SfSelect,
     SfBreadcrumbs,
     LazyHydrate,
-    SfLink,
     SfLoader,
     SfProductCardHorizontal,
+    SfAccordion,
+    SfChevron,
   },
   setup(_, context) {
-    const { getFacetsFromURL, getCatLink, getProductLink, changeSorting, changeFilters } =
+    const { getFacetsFromURL, getProductLink, changeSorting, changeFilters, setCategoryLink } =
       useUiHelpers()
     const { result, search, loading } = useFacet(`category-listing`)
-    const facetsFromUrl = ref({ sort: "" })
     const nuxt = useNuxtApp()
     const { sortOptions } = nuxt.nuxt2Context.$config.productListing
+    const facetsFromUrl = ref({
+      categoryCode: "",
+      page: null,
+      itemsPerPage: 20,
+      phrase: "",
+      filters: [],
+      sort: "",
+    })
+    const isSearchPage = ref(false)
+    const route = useRoute()
+
+    // Determining if search page using categoryCode present in URL or not
+    if (!route.value.params?.categoryCode) {
+      isSearchPage.value = true
+    }
 
     const {
       result: productSearchResult,
       search: productSearch,
       loading: productSearchLoading,
     } = useProductSearch(`product-search`)
 
-    const route = useRoute()
+    useAsync(async () => {
+      facetsFromUrl.value = getFacetsFromURL(isSearchPage.value)
+      await search(facetsFromUrl.value)
+      await productSearch(facetsFromUrl.value)
+    }, null)
+
+    const breadcrumbs = computed(() => facetGetters.getBreadcrumbs(result?.value))
     const products = computed(() => productSearchGetters.getProducts(productSearchResult?.value))
     const facets = computed(() =>
       productSearchGetters.getFacets(productSearchResult?.value, ["Value", "RangeQuery"])
     )
-    const categoryTree = computed(() => facetGetters.getCategoryTree(result?.value))
-    const breadcrumbs = computed(() => facetGetters.getBreadcrumbs(result?.value))
-    const categoryName = computed(() => categoryGetters.getName(categoryTree.value?.[0]))
-    const categoryTitle = computed(() => categoryGetters.getName(categoryTree.value?.[0]))
-    const childrenCategories = computed(() => categoryTree.value?.[0]?.childrenCategories)
-    const navCategories = useState("nav-categories", () => [])
-    const showMoreButton = useState("show-more-button", () => false)
-
-    const visibleCategories = (categories, categoriesVisible = 5) => {
-      showMoreButton.value = !showMoreButton.value
-      navCategories.value = categories.slice(0, categoriesVisible)
-    }
-
-    const selectFilter = (filterValue) => {
-      const qs = route.value?.query as { filters: string }
-      const filters = qs.filters?.split(",") || []
-      const currentIndex = filters.indexOf(filterValue)
-      if (currentIndex > -1) {
-        filters.splice(currentIndex, 1)
+    const categoriesFromSearch = computed(() => {
+      let header: string, children: FacetValue[]
```

Comment:
move this function to productSearchGetter, getCategoryFacet that returns the { header, children }

Reply:
Done